### DISCUSSION TITLE
VSCode formatting extension for pg-promise sql files that includes Named Parameters

### DISCUSSION THREAD

**Discussion by KostiantynO** [vitaly-t/pg-promise#872] [Q&A] [answered]
Hello!
  What VSCode extention do you use to format SQL files, while working with 'pg-promise' Named Parameters?

I have an issue - cannot format the *.sql files, that contain special templates from 'pg-promise', like 
```sql 
${user_id}
```
 - gets formatted on save to 
```sql
$ { user_id }
```
, which breaks the template literal.

**Comment by vitaly-t** [comment]
For development, I use IntelliJ IDEA, and whatever plug-in it uses by default for SQL (too many of them). It never breaks any SQL code.

I don't know which plugin works correctly under VSCode, but clearly the one you are using is not good.

**Comment by vitaly-t** [comment, marked as answer]
P.S. Later on, I tried SQL Beautify extension for VSC, and it does not break variables.

**Reply by KostiantynO** [reply]
Do you mean this one? [SQL Beautify](https://marketplace.visualstudio.com/items?itemName=clarkyu.vscode-sql-beautify)
✅ Will use it then, thank you!  And it works with Named Parameters!
  Select text -> and press `ctrl + shift + f`
